commandbar.d.ts

```diff
@@ -1,99 +1,125 @@
-    type: 'dependent';
-    value: string;
-    order_key: number;
-};
+    export enum CommandBarArgumentTypes {
+      context = "context",
+      provided = "provided",
+      dependent = "dependent",
+      function = "function",
+      set = "set",
+    }
 
-export type FunctionArgument = {
-    type: 'function';
-    value: string;
-    order_key: number;
-};
+    export type ContextArgument = {
+      type: CommandBarArgumentTypes.context;
+      value: string;
+      order_key: number;
+      label_field?: string;
+    };
 
-export type SetArgument = {
-    type: 'set';
-    value: string[];
-    order_key: number;
-};
+    export type ProvidedArgument = {
+      type: CommandBarArgumentTypes.provided;
+      value: "text" | "time";
+      order_key: number;
+    };
 
-export type CommandBarArgument =
-    | ContextArgument
-    | ProvidedArgument
-    | DependentArgument
-    | FunctionArgument
-    | SetArgument;
+    export type DependentArgument = {
+      type: CommandBarArgumentTypes.dependent;
+      value: string;
+      order_key: number;
+    };
 
-export type CommandBarCommandConfig = {
-    name: string;
-    text: string;
-    template: LinkCommandTemplate | CallbackCommandTemplate;
-    shortcut_mac?: string[];
-    shortcut_win?: string[];
-    tags?: string[];
-    explanation?: string;
-    sort_key?: number;
-    arguments?: Record<string, CommandBarArgument>;
-};
+    export type FunctionArgument = {
+      type: CommandBarArgumentTypes.function;
+      value: string;
+      order_key: number;
+    };
 
-export interface CommandBar {
-    readonly boot: (args: CommandBarBootArgs) => void;
-    readonly addContext: (context: CommandBarContext) => void;
-    readonly removeContext: (key: string) => void;
-    readonly setContext: (context: CommandBarContext) => void;
-    readonly addCallback: (name: string, callback: CommandBarCallback | undefined) => void;
-    readonly removeCallback: (name: string) => void;
-    readonly addCommand: (command: CommandBarCommandConfig) => void;
-    readonly removeCommand: (name: string) => void;
-    readonly addRouter: (routerFunc: CommandBarRouterFunc) => void;
-    readonly addSearch: (category: string, searchFunc: CommandBarSearchFunc) => void;
-    readonly showGuide: (key: string) => void;
-    readonly open: (input?: string) => void;
-    readonly execute: (command: number | string) => void;
-    readonly onboard: () => void;
-    readonly isOpen: () => boolean | undefined;
-}
+    export type SetArgument = {
+      type: CommandBarArgumentTypes.set;
+      value: string[];
+      order_key: number;
+    };
 
-declare global {
-    interface Window {
-        CommandBar: CommandBar;
-    }
+    export type CommandBarArgument =
+      | ContextArgument
+      | ProvidedArgument
+      | DependentArgument
+      | FunctionArgument
+      | SetArgument;
+
+    export type CommandBarCommandConfig = {
+      name: string;
+      text: string; // React.ReactNode type possible?
```

Comment:
`React.ReactNode` isn't possible yet, but we will probably add support for it soon.

Reply:
@jamesaevans Removed that comment